### DISCUSSION TITLE
custom port?

### DISCUSSION THREAD

**Discussion by sie-sh** [coddingtonbear/obsidian-web#131] [Ideas]
Usually run multiple Obsidian vaults at once, all running their own REST API install with a custom port. I'd really appreciate the ability to have custom configurations, or at least to change the port in the clipper settings. Thanks!

**Comment by zsgcgrs** [comment]
A custom port is needed to connect to the api interface behind the reverse proxy for https connection

**Reply by coddingtonbear** [reply]
I'm not sure I follow, @zsgcgrs, I regularly use HTTPS via a reverse proxy without needing to specify a custom port.

**Reply by zsgcgrs** [reply]
I tried to use traefik reverse proxy http://127.0.0.1:27123 and https://127.0.0.1:27124, using the custom port 8443. When filling in example.com:8443 in the plug-in, it will prompt example.com:8443 :27123, when filling in https://example.com:8443, there is no prompt about 27123, but an error is still reported and the connection cannot be made.

**Reply by coddingtonbear** [reply]
I see -- what is preventing you from using the standard port for the reverse proxy?

**Reply by zsgcgrs** [reply]
Because the home broadband in my country has disabled ports 80 and 443, I can only use non-standard ports. If the next version can accept non-standard ports, I should go to reverse proxy when doing reverse proxy http://127.0.0.1:27123 still https://127.0.0.1:27124

**Reply by coddingtonbear** [reply]
By "standard ports", I mean using the standard ports used by Obsidian Local REST API (ie `27123` and `27124`).

**Reply by zsgcgrs** [reply]
Because all of my NAS uses traefik reverse proxy to access various services using a second level domain name and a unified port, and it is also convenient to configure SSL to ensure security

**Reply by coddingtonbear** [reply]
I suppose I still don't follow -- you clearly have to be setting a port for the reverse proxy. Could you elaborate on why the port needs to differ from 27123 and 27124?  The reason I'm asking is that this is sounding more and more like a problem very specific to how you personally have set up your reverse proxy, and I'm afraid that the bar that needs to be met for me to consider allowing extra complexity into this project is quite a lot higher than that.

**Reply by zsgcgrs** [reply]
I just don’t want to open an extra port on the router to implement this reverse proxy.
